fhirpipe/references/reference_binder.py

```diff
@@ -0,0 +1,123 @@
+import re
+import logging
+from collections import defaultdict
+
+import fhirpipe
+from fhirpipe.load.fhirstore import get_mongo_client, get_resource_instances
+
+
+class ReferenceBinder:
+    def __init__(self, fhirstore, skip_ref_binding=False):
+        self.fhirstore = fhirstore
+        self.skip_ref_binding = skip_ref_binding
+
+        # store of the form
+        # {
+        #   fhir_type: {(value, system): fhir_id, ...},
+        #   fhir_type: {(value, system): fhir_id, ...},
+        #   ...
+        # }
+        self.indentifiers_store = {}
+
+        # map from resource_id to resource type
+        self.map_resource_type = {}
+
+        # map_resource_references is a dict of the form
+        # {resource_id: [reference_path, ...], ...}
+        self.map_resource_references = defaultdict(list)
+
+    def bind_references(self):
+        if self.skip_ref_binding:
```

Comment:
I think this is pointless to refer to this option directly in the ReferenceBinder class (see comment below)

Reply:
Yeah, I know. It was in the run function before every line having to do with binding before but it was also ugly and it made the run() function complexity > 10...
What do you think I should do?

Reply:
yeah ok leave it here, it's not the best but meh